cfme/storage/object_store_container.py

```diff
@@ -0,0 +1,166 @@
+    toolbar = View.nested(ObjectStoreContainerToolbar)
+    including_entities = View.include(BaseEntitiesView, use_parent=True)
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_container and
+            self.title.text == 'Cloud Object Store Containers')
+
+
+class ObjectStoreContainerDetailsView(ObjectStoreContainerView):
+    """The detail Object Store containers page"""
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].key)
+
+        return (
+            self.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(ObjectStoreContainerDetailsToolbar)
+    sidebar = View.nested(ObjectStoreContainerDetailsAccordion)
+    entities = View.nested(ObjectStoreContainerDetailsEntities)
+
+
+class ObjectStoreContainerCollection(BaseCollection):
+    """Collection object for :py:class:'cfme.storage.object_store_container.ObjectStoreContainer'
+    """
+
+    def __init__(self, appliance):
+        self.appliance = appliance
+
+    def instantiate(self, key, provider):
+        return ObjectStoreContainer(self, key, provider)
+
+    def all(self, provider):
+        """returning all containers objects"""
+        view = navigate_to(self, 'All')
+        containers = [self.instantiate(key=item.name, provider=provider)
+                 for item in view.entities.get_all()]
+        return containers
+
+
+class ObjectStoreContainer(BaseEntity, WidgetasticTaggable):
+    """ Model of an Storage Object Store Containers in cfme
+
+    Args:
+        key: key of the container.
+        provider: provider
+    """
+    def __init__(self, collection, key, provider):
+        self.collection = collection
+        self.appliance = self.collection.appliance
+        self.key = key
+        self.provider = provider
+
+    # TODO add create method after BZ 1490320 fix
+
+
+@navigator.register(ObjectStoreContainerCollection, 'All')
+class All(CFMENavigateStep):
+    VIEW = ObjectStoreContainerAllView
+    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+
+    def step(self):
+        if self.obj.appliance.version < "5.8":
+            self.prerequisite_view.navigation.select('Storage', 'Object Stores')
+        else:
+            self.prerequisite_view.navigation.select(
+                'Storage', 'Object Storage', 'Object Store Containers')
+
+    def resetter(self):
+        self.view.toolbar.view_selector.select("Grid View")
+
+
+@navigator.register(ObjectStoreContainer, 'Details')
+class Details(CFMENavigateStep):
+    VIEW = ObjectStoreContainerDetailsView
+    prerequisite = NavigateToAttribute('collection', 'All')
+
+    def step(self, *args, **kwargs):
+        try:
+            self.prerequisite_view.entities.get_entity(self.obj.key,
+                                                       surf_pages=True).click()
+        except NoSuchElementException:
```

Comment:
REQUIRED: There shouldn't be NoSuchElementException. Please double check this. As far as I remember get_entity returns ItemNotFound in this case. I don't think you need to handle it.

Reply:
I am going to recheck it. thank you